### ISSUE TITLE
Reader: improve accessibility of add new list

### ISSUE THREAD

**Issue by bluefuton** [Automattic/wp-calypso#5047]
The sidebar form to add new lists is not a great experience when using a screen reader.

The form input does not have a label, and a user has pointed out:

> JAWS reads out the Add button before the edit field for the list name.

Could we potentially open the edit form when creating a new list, rather than trying to create it in the sidebar?

**Comment by bluefuton** [comment]
It's no longer possible to create a new list from the sidebar.

An `aria-label` was added in #6338, which has improved the similar process of adding a new tag.
